### DISCUSSION TITLE
MSAL Example for Web API with Flask

### DISCUSSION THREAD

**Discussion by barmettlerl** [AzureAD/microsoft-authentication-library-for-python#499] [General]
While searching through the several examples, I saw that there is no Use-Case for MSAL for a Web API and Flask. Is this setup even supported or planned to be supported by MSAL Python. 

Thanks

**Comment by r-leyshon** [comment]
Yes it is strange that the [official docs](https://learn.microsoft.com/en-us/entra/identity-platform/sample-v2-code?tabs=apptype) do not reference it. Does anyone have an equivalent implementation for the [optionally call Microsoft Graph API tutorial](https://github.com/Azure-Samples/ms-identity-python-webapp). The closest example I see is [here](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/sample/interactive_sample.py)

**Comment by bgavrilMS** [comment]
Hmm, I also can't find the web api sample. @rayluo - do we have any? There don't seem to be any listed as official samples https://learn.microsoft.com/en-us/entra/identity-platform/sample-v2-code?tabs=apptype

In the meantime, you can follow the NodeJS web api sample, it should be fairly similar - https://github.com/Azure-Samples/ms-identity-javascript-tutorial/tree/main/3-Authorization-II/1-call-api

Just note that Microsoft does not produce any token validation libraries except for .NET. General Purpose jwt token validation libraries should work fine in single tenant scenarios. It's more complex for multi-tenant scenarios. 

CC @rayluo @localden
